src/test/integration/integration-test.py

```diff
@@ -0,0 +1,102 @@
+            cls.session = utils.setup_session(config)
+            cls.test_cases = config['test_cases']
+
+        with open(openapi_path) as openapi_file:
+            cls.openapi = yaml.load(openapi_file)
+
+    @classmethod
+    def cleanup(cls):
+        cls.session.close()
+
+    # Test case: GET /spendingauthority with onid filter
+    def test_get_onids_with_filter(self, endpoint='/spendingauthority'):
+        testing_authority_onids = ['kuok', 'wetherel'] # Valid onids with spending authority
+        testing_non_authority_onids = ['alawammo', 'wilsonai'] # Valid onids with no spending authority
+        testing_bad_request = [''] # Invalid request
+
+        for onid in testing_authority_onids:
+            params = {'onid': onid}
+            response = utils.make_request(self, endpoint, 200, params=params)
+            spending_schema = utils.get_resource_schema(
+                self, 'SpendingAuthorityResource'
+            )
+
+            utils.check_schema(self, response, spending_schema)
+
+            response_data = response.json()['data']
+            actual_onid = response_data['id']
+            self.assertEqual(actual_onid.lower(), onid.lower())
+
+        for onid in testing_non_authority_onids:
+            params = {'onid': onid}
+            response = utils.make_request(self, endpoint, 200, params=params)
+            spending_schema = []
+            try:
+                self.assertEqual(response.json()['data'], spending_schema)
+            except AssertionError:
+                logging.warning('data should be an empty array')
+
+
+        # Testing Error response
+        for onid in testing_bad_request:
+            params = {'onid': onid}
+            response = utils.make_request(self, endpoint, 400, params=params)
+            spending_schema = utils.get_resource_schema(self, 'Error')
+            utils.check_schema(self, response, spending_schema)
+
+        # Checking that the lists of spending limits and indexes are unique
+        # Checking valid onids with spending authority only
+        for onid in testing_authority_onids:
+            params = {'onid': onid}
+            response = utils.make_request(self, endpoint, 200, params=params)
+            spending_schema = utils.get_resource_schema(
+                self, 'SpendingAuthorityResource'
+            )
+
+            response_data = response.json()['data']
+            attributes = response_data['attributes']
+            limits = attributes['limits']
+            indexes = []
+            spendingLimit = []
+
+            # Checking that each spending limit has at least one index
+            for i in limits:
+                indexes.append(i['indexes'])
+                spendingLimit.append(i['spendingLimit'])
+                self.assertGreater(len(i['indexes']), 0)
+
+            # Creating a dict indexes_list from indexes, and compare it with the original array of indexes
+            for i in indexes:
```

Comment:
We discussed this in person, but this loop does not test that "a given index appears in the whole response once". My recommendation would be making the indexes object a 1 dimensional array in the parent loop, and then checking for duplicate entries here.

Reply:
Another suggestion, when using `i` as a variable name, it usually meant to be an "index". To avoid the confusion, can you name them more descriptive?